pyblazing/pyblazing/apiv2/context.py

```diff
@@ -1390,18 +1394,19 @@ def _optimize_with_skip_data_getSlices(self, current_table, scan_table_query,sin
             uri_values = []
             row_groups_ids = []
 
-            if not file_indices_and_rowgroup_indices.empty: #skipdata did not filter everything
-                file_and_rowgroup_indices = file_indices_and_rowgroup_indices.to_pandas()
-                files = file_and_rowgroup_indices['file_handle_index'].values.tolist()
-                grouped = file_and_rowgroup_indices.groupby('file_handle_index')
-                for group_id in grouped.groups:
-                    row_indices = grouped.groups[group_id].values.tolist()
-                    actual_files.append(current_table.files[group_id])
-                    if group_id < len(current_table.uri_values):
-                        uri_values.append(current_table.uri_values[group_id])
-                    row_groups_col = file_and_rowgroup_indices['row_group_index'].values.tolist()
-                    row_group_ids = [row_groups_col[i] for i in row_indices]
-                    row_groups_ids.append(row_group_ids)
+            for file_indices_and_rowgroup_indices_item in file_indices_and_rowgroup_indices:
```

Comment:
We should make `runSkipDataCaller` return the same as it used to and not implement all these changes here. We are always expecting the output to have one partition. This code which assumes otherwise just makes things more complicated

Reply:
I removed that code. The modifications now are only for the runQuery function.